code/modules/mob/abstract/new_player/character_traits.dm

```diff
@@ -79,3 +79,87 @@
 	name = "Major Hemophilia"
 	desc = "Your blood lacks ALL clotting factors, causing wounds to never stop bleeding."
 	trait_type = TRAIT_DISABILITY_HEMOPHILIA_MAJOR
+
+
+ABSTRACT_TYPE(/datum/character_disabilities/bruising)
+	name = "Bruised Limb"
+	var/affected_limb
+
+/datum/character_disabilities/bruising/apply_self(var/mob/living/carbon/human/target)
+	target.apply_damage(10, DAMAGE_BRUTE, affected_limb, armor_pen = 100, silent = TRUE)
+
+#define BRUISING_DISABILITY(LIMB_PATH, LIMB_NAME, LIMB_TAG) \
```

Comment:
I don't think this is a good pattern or use case for a macro, it should just list the types and build the name on `New()` based on the name of the zone it's targeting

Reply:
i don't really agree there, if i build it on "New", it'd have to be arbitrarily put into the global disabilities for loop for every different tyoe, making that less readible

the reason both limb name and limb tag is defined here is because, while BP_GROIN translates perfectly to the english word "Groin", BP_R_LEG translates to "r_leg", so manually defining it is just easier

see escape_pods.dm for precedent